src/pipeline/mod.rs

```diff
@@ -0,0 +1,74 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use datafusion::prelude::{SessionConfig, SessionContext};
+
+use crate::{
+    QueryDialect,
+    io_strategies::{input_strategy::InputStrategy, output_strategy::OutputStrategy},
+    operations::data_operation::DataOperation,
+};
+
+#[derive(Default)]
+pub struct PipelineConfig {
+    pub temp_dir: Option<PathBuf>,
+}
+
+pub struct Pipeline {
+    input_strategy: InputStrategy,
+    operations: Vec<Box<dyn DataOperation>>,
+    output_strategy: OutputStrategy,
+    config: PipelineConfig,
+}
```

Comment:
this is the main abstraction, a pipeline of InputStrategy => 1+ DataOperations => OutputStrategy